### PR TITLE
[BD-18] switch from nose to pytest

### PR DESCRIPTION
**Pull request by emZubair**
## [BD-18] switch from nose to pytest
Updated Makefile commands as per pytest options.

**Packages Added**

- pytest-cov

**Packages Removed**

- nose
- nose-ignore-docstring

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -103,8 +103,9 @@ quality-docker:
 coverage-docker:
 	docker run --rm -u root -v `(pwd)`:/edx/app/analytics_pipeline/analytics_pipeline -it edxops/analytics_pipeline:latest coverage xml
 
-coverage-local: test-local
```
Comment (ziafazal) [inline Makefile:106]: why we removed `test-local` here?

File: Makefile [hunk 2]
```diff
@@ -103,8 +103,9 @@ quality-docker:
...
+coverage-local:
 	python -m coverage html
```
Comment (ziafazal) [inline Makefile]: I think we can use single command to accomplish what all these `coverage` commands are doing except the `diff-cover`. Could you please try?
`coverage run -m pytest html xml -o coverage.xml`

**Review by ziafazal — CHANGES_REQUESTED** [review]
@emzubair could you please try my suggestion?

File: Makefile (hunk 1 above)
Comment (emZubair) [reply Makefile:106]: it was just of local testing purposes, reverting the change.

**Review by ziafazal — APPROVED** [review]
👍

**Pull request closed without merging**